packages/harmony/src/components/popup/Popup.tsx

```diff
@@ -189,8 +190,44 @@ const closingAnimationDuration = 90
 export const Popup = forwardRef<HTMLDivElement, PopupProps>(function Popup(
   props: PopupProps,
   ref: Ref<HTMLDivElement>
+) {
+  const [popupState, setPopupState] = useState<ModalState>('closed')
+
+  const isVisible = popupState !== 'closed'
+  useEffect(() => {
+    if (popupState === 'closed' && props.isVisible) {
+      setPopupState('opening')
+    } else if (popupState === 'open' && !props.isVisible) {
+      setPopupState('closing')
+    }
+  }, [props.isVisible, popupState])
```

Comment:
nit: maybe destructure this - `const { isVisible: isVisibleProp } = props`

Reply:
not sure i understand why, but done 👍